vllm/core/block_manager.py

```diff
@@ -170,12 +219,50 @@ def can_append_slot(self, seq_group: SequenceGroup) -> bool:
         num_seqs = seq_group.num_seqs(status=SequenceStatus.RUNNING)
         return num_seqs <= num_free_gpu_blocks
 
-    def append_slot(self, seq: Sequence) -> Optional[Tuple[int, int]]:
+    def _promote_last_block(self, seq: Sequence, block: PhysicalTokenBlock):
+        # Compute a new hash for the block so that it can be shared by other Sequences
+        new_hash = seq.hash(len(seq.logical_token_blocks) - 1)
+
+        # TODO: What if the hash already exists in the table? If it does, we can free and use that block?
+        self.gpu_allocator.update_hash(new_hash, block)
```

Comment:
If hash already exists, then we should indeed free and discard the current block

Reply:
Done! Thanks for the review comment.